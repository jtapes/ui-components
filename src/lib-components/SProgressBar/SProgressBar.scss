@import '../../theme/styles/variables';
@import '../../theme/styles/mixins';
@import '../../theme/styles/reset';
@import '../../theme/styles/fonts';
@import '../../theme/styles/global';

$progress-bar-active-color: $color--success !default;
$progress-bar-height: $size--4 !default;
$progress-bar-bgColor: $color--border !default;

.s-progress-bar {
  &__line {
    height: $progress-bar-height;
    background: $progress-bar-bgColor;
  }

  &__progress {
    height: $progress-bar-height;
    background: $progress-bar-active-color;
  }
}
