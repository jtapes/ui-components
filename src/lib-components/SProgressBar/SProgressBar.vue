<template>
  <div class="s-progress-bar">
    <div class="s-progress-bar__above">
      <slot name="above" />
    </div>
    <div class="s-progress-bar__line">
      <div ref="progressLine" class="s-progress-bar__progress" />
    </div>
    <div class="s-progress-bar__under">
      <slot name="under" />
    </div>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop, Watch } from 'vue-property-decorator'

@Component
export default class SProgressBar extends Vue {
  @Prop({
    type: Number,
    validator: (val: number) => val >= 0 && val <= 100,
    default: 0,
  })
  progress!: number

  $refs!: {
    progressLine: HTMLDivElement
  }

  @Watch('progress')
  onProgressChanged(newProgressValue: number): void {
    this.updateProgress(newProgressValue)
  }

  updateProgress(val = this.progress): void {
    this.$refs.progressLine.style.width = `${val}%`
  }

  mounted(): void {
    this.updateProgress()
  }
}
</script>
