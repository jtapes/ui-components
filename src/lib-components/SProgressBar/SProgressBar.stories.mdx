import { config, withDesign } from 'storybook-addon-designs'
import { withTests } from '@storybook/addon-jest';
import {
  Meta,
  Story,
  ArgsTable,
  Description,
  Source,
  Preview,
} from '@storybook/addon-docs'
import results from '/.jest-test-results.json';
import SProgressBar from '@/lib-components/SProgressBar/SProgressBar.vue'

<Meta
  title="DESIGN SYSTEM/Atom/ProgressBar"
  component={SProgressBar}
  decorators={[withDesign, withTests({results})]}
  argTypes={{
    progress: { control: 'range' },
  }}
  parameters={{
    design: config({
      type: 'figma',
      url:
        'https://www.figma.com/file/42Xh2qI3qvq2CRkc2HJlwA/MVP-3-Market-Catalog-%E2%80%A2-Product-Page?node-id=1128%3A88',
    }),
    docs: {
      source: {
        type: 'code',
      },
    },
  }}
/>

# Компонент визуально отображающий прогресс от 0 до 100%

<Description of={SProgressBar} />

export const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { SProgressBar },
  template:
    `<s-progress-bar :progress="progress" />`,
})

<Preview withToolbar withSource="none">
  <Story name="SProgressBar" args={{ progress: 25.32 }}>
    {Template.bind({})}
  </Story>
</Preview>

## Пример

<Source
  language="html"
  code={`
    <s-progress-bar :progress="25.32" />
  `}
/>

<ArgsTable of={SProgressBar} />

