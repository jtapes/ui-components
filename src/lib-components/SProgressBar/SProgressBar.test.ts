import { mount } from '@vue/test-utils'
import SProgressBar from '@/lib-components/SProgressBar/SProgressBar.vue'

describe('SProgressBar', () => {
  test('Is Vue snapshot', () => {
    const wrapper = mount(SProgressBar, {
      propsData: {
        progress: 25,
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  test('Test real width of bar', () => {
    const wrapper = mount(SProgressBar, {
      propsData: {
        progress: 25,
      },
    })
    const progressBar = wrapper.find('.s-progress-bar__progress')
    expect((progressBar.element as HTMLElement).style.width).toEqual('25%')
  })
})
