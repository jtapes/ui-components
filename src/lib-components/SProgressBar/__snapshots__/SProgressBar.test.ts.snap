// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SProgressBar Is Vue snapshot 1`] = `
<div
  class="s-progress-bar"
>
  <div
    class="s-progress-bar__above"
  />
   
  <div
    class="s-progress-bar__line"
  >
    <div
      class="s-progress-bar__progress"
      style="width: 25%;"
    />
  </div>
   
  <div
    class="s-progress-bar__under"
  />
</div>
`;
